scm/driver/azure/azure.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2017 Drone.IO Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package azure implements a azure client.
+package azure
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/drone/go-scm/scm"
+)
+
+// New returns a new azure API client.
+func New(uri, azureOwner, azureProject string) (*scm.Client, error) {
+	base, err := url.Parse(uri)
+	if err != nil {
+		return nil, err
+	}
+	if !strings.HasSuffix(base.Path, "/") {
+		base.Path = base.Path + "/"
+	}
+	if azureOwner == "" || azureProject == "" {
+		return nil, fmt.Errorf("azure owner and azure project are required")
+	}
+	client := &wrapper{
+		new(scm.Client),
+		azureOwner,
+		azureProject,
+	}
+	client.BaseURL = base
+	// initialize services
+	client.Driver = scm.DriverAzure
+	// client.Linker = &linker{websiteAddress(base)}
+	client.Contents = &contentService{client}
+	// client.Git = &gitService{client}
+	// client.Issues = &issueService{client}
+	// client.Organizations = &organizationService{client}
+	// client.PullRequests = &pullService{client}
+	// client.Repositories = &RepositoryService{client}
+	// client.Reviews = &reviewService{client}
+	// client.Users = &userService{client}
+	// client.Webhooks = &webhookService{client}
+	return client.Client, nil
+}
+
+// NewDefault returns a new azure API client.
+func NewDefault(azureOwner, azureProject string) *scm.Client {
+	client, _ := New("https://dev.azure.com", azureOwner, azureProject)
+	return client
+}
+
+// wrapper wraps the Client to provide high level helper functions for making http requests and unmarshaling the response.
+type wrapper struct {
+	*scm.Client
+	AzureOwner   string
+	AzureProject string
```

Comment:
could these be changed to private variables?

Reply:
done